src/lib/externaldata.py

```diff
@@ -93,6 +94,64 @@ def to_json(self):
 
         return json.dumps(json_data, indent=4)
 
+
+class ExternalDataSource(ExternalData):
+    def __init__(self, source, data_type, url):
+        name = source.get('filename')
+        if not name:
+            name = source.get('dest-filename')
+        if not name:
+            name = os.path.basename(url)
```

Comment:
You could do all these with `name = result1 or result2 or result3`, right?

Reply:
Yep, this looks nicer; good point.